musmart/timing/tempo_modeling.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""This module implements various functions related to linear tempo modeling e.g., `tempo_slope`, `tempo_drift`"""
+
+from typing import Iterable
+
+import numpy as np
+
+
+__all__ = ["tempo_slope", "tempo_drift", "tempo_fluctuation"]
+__author__ = "Huw Cheston"
+
+
+def _fit_tempo_linregress(beats: Iterable):
+    """Fits linear regression of BPM measurements vs onset time"""
+
+    from scipy.stats import linregress
+    # Dependent variable: BPM measurements
+    y = np.array(60 / np.diff(beats))
+    # Predictor variable: the onset time
+    x = np.array(beats[1:])
+    # Handling NaNs: remove missing values from both arrays
```

Comment:
I don't think you should silently remove NaNs. Why would they occur? Presumably when `np.diff(beats)` returns 0. But why would that happen?